.github/workflows/test.yml

```diff
@@ -43,9 +43,15 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         python -m pip install numpy scipy cython sphinx matplotlib numpydoc sphinx_rtd_theme
-        pip install git+https://github.com/qutip/qutip.git
+        pip install qutip
         pip install .
-    - name: Test with pytest
+    - name: Test code snippets in the documentation
       run: |
         cd docs
         make doctest
+    - name: Test code examples for the pulse paper
+      run: |
+        python -m pip install joblib pytest pytest-custom_exit_code
+        cd docs/pulse-paper
+        mkdir figures
+        pytest *.py --suppress-no-test-exit-code
```

Comment:
It runs the snippets, if any of the code produces an error, it will fail, as shown in many of my previous attempts, e.g., https://github.com/qutip/qutip-qip/runs/2614682544?check_suite_focus=true 

The `--suppress-no-test-exit-code` option exists because `pytest` doesn't treat any of the code snippets as a formally defined tests (There is no assert anyway). It just runs through the code and ends if everything is fine.